### PR TITLE
Allow Super admins to Archive and Merge users

### PR DESCRIPTION
**Pull request by xsrust**
Adds 2 options to the super-admin user-edit page:
`Archiving` will remove the user's name, email, and sign-in-IPs.
`Merging` will swap apply the currently edited user's ID to all of the selected user's notes, plans, and answers, then remove the selected user.

Screenshot 2019-09-04 at 11 54 14

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -326,10 +327,7 @@ def remove_token!
   # Returns Boolean
   def keep_or_generate_token!
     if api_token.nil? || api_token.empty?
-      new_token = loop do
-        random_token = SecureRandom.urlsafe_base64(nil, false)
-        break random_token unless User.exists?(api_token: random_token)
-      end
+      new_token = User.unique_random(field_name: 'api_token')
```
Comment (briri) [inline app/models/user.rb]: nice. it seems more logical to place that functionality on the User model

File: app/models/user.rb [hunk 2]
```diff
@@ -391,6 +389,41 @@ def acknowledge(notification)
...
+    recovery_email = nil
+    api_token = nil
+    encrypted_password = nil
+    last_sign_in_ip = nil
+    current_sign_in_ip =  nil
+    active = false
+    return self.save
+  end
+
+  def merge(to_be_merged)
```
Comment (briri) [inline app/models/user.rb]: nice and clean. 👍

**Comment by xsrust** [comment]
Thanks @briri 
I started by using a typeahead, but it was not too happy loading with all of our (32k) emails on my dev machine.  
I figured since it's only for super-admins at the moment, and they'll probably have the full email anyway, that this would do for a first-pass.

**Comment by xsrust** [comment]
That being said, If we were to open it up to Org-admins, scoping it by their organisation should solve that issue of too many user accounts

**Comment by briri** [comment]
makes sense @xsrust

**Pull request merged**